thewalrus/tests/test_quantum.py

```diff
@@ -674,3 +676,16 @@ def test_is_classical_cov_thermal(nbar):
     """ Tests that a thermal state is classical"""
     cov = (2 * nbar + 1) * np.identity(2)
     assert is_classical_cov(cov)
+
+
+def test_total_photon_num_dist_pure_state():
+    """ Test the correct photon number distribution is obtained for n modes
+    with nmean number of photons up to Fock cutoff nmax"""
+    n = 3
+    nmean = 1.0
+    nmax = 50
+    rs = np.arcsinh(np.sqrt(nmean)) * np.ones([n])
+    cov = np.diag(np.concatenate([np.exp(2 * rs), np.exp(-2 * rs)]))
+    p1 = total_photon_num_dist_pure_state(cov, nmax=nmax)
+    p2 = gen_single_mode_dist(np.arcsinh(np.sqrt(nmean)), N=n, nmax=nmax)
+    assert np.allclose(p1, p2)

```

Comment:
do we also need a test for `gen_multi_mode_dist`?

Reply:
It is implicitly tested when `total_photon_num_dist_pure_state` is called. In fact this last function is just a wrapper that extracts the squeezing parameters from a pure covariance matrix.